### PR TITLE
Clarify fed2 composition rules for enums

### PR DESCRIPTION
**Pull request by StephenBarlow**

### CODE REVIEW

File: docs/source/federated-types/composition.mdx [hunk 1]
```diff
@@ -365,18 +365,21 @@ As you can see, the supergraph schema includes only the input fields and argumen
...
-graph TB;
-    enum("Is the enum used as the type for at least<br/>one field argument or input type field?");
-    enum--No-->union("Union");
-    enum--Yes-->intersection("Intersection");
-```
+| Scenario | Strategy |
+|----------|----------|
+| The enum is used as the return type for at least one object or interface field. | Union |
+| The enum is used as the type for at least one field argument or input type field. | Intersection |
+| **Both** of the above are true. | All definitions must match **exactly**. |
```
Comment (rkoron007) [inline docs/source/federated-types/composition.mdx:374]: I like clarifying that `enum` definitions must match: 
```suggestion
| **Both** of the above are true. | All `enum` definitions must [match **exactly**](#exact-match). |
```

File: docs/source/federated-types/composition.mdx [hunk 2]
```diff
@@ -365,18 +365,21 @@ As you can see, the supergraph schema includes only the input fields and argumen
...
 
-```mermaid
-graph TB;
-    enum("Is the enum used as the type for at least<br/>one field argument or input type field?");
-    enum--No-->union("Union");
-    enum--Yes-->intersection("Intersection");
-```
+| Scenario | Strategy |
+|----------|----------|
+| The enum is used as the return type for at least one object or interface field. | Union |
```
Comment (rkoron007) [inline docs/source/federated-types/composition.mdx:372]: Adding links to this table might help people navigate a little easier:
```suggestion
| The enum is used as the return type for at least one object or interface field. | [Union](#union) |
```

File: docs/source/federated-types/composition.mdx [hunk 3]
```diff
@@ -365,18 +365,21 @@ As you can see, the supergraph schema includes only the input fields and argumen
...
+| The enum is used as the type for at least one field argument or input type field. | Intersection |
```
Comment (rkoron007) [inline docs/source/federated-types/composition.mdx:373]:
```suggestion
| The enum is used as the type for at least one field argument or input type field. | [Intersection](#intersection) |
```

**Review by rkoron007 — APPROVED** [review]
Looks good to me! I added a few small suggestions, but feel free to take/leave them!

File: docs/source/federated-types/composition.mdx (hunk 1 above)
Comment (rkoron007) [inline docs/source/federated-types/composition.mdx:374]:
```suggestion
| **Both** of the above are true. | All definitions must [match **exactly**](#exact-match) |
```

**Review by benweatherman — APPROVED** [review]
🐐

**Comment by jeffjakub** [comment]
This looks great, thanks Stephen.

**Comment by StephenBarlow** [comment]
Thanks all!

**Pull request merged**